spot_driver_cpp/src/interfaces/rclcpp_parameter_interface.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/interfaces/rclcpp_parameter_interface.hpp>
+
+#include <cstdlib>
+
+namespace
+{
+constexpr auto kEnvVarNameAddress = "SPOT_IP";
+constexpr auto kEnvVarNameUsername = "BOSDYN_CLIENT_USERNAME";
+constexpr auto kEnvVarNamePassword = "BOSDYN_CLIENT_PASSWORD";
+
+constexpr auto kParameterNameAddress = "address";
+constexpr auto kParameterNameUsername = "username";
+constexpr auto kParameterNamePassword = "password";
+constexpr auto kParameterNameRGBImageQuality = "image_quality";
+constexpr auto kParameterNameHasRGBCameras = "rgb_cameras";
+constexpr auto kParameterNamePublishRGBImages = "publish_rgb";
+constexpr auto kParameterNamePublishDepthImages = "publish_depth";
+constexpr auto kParameterNamePublishDepthRegisteredImages = "publish_depth_registered";
```

Comment:
These are specifically used for declaring ROS 2 parameters, so I think it makes sense to keep them with the rclcpp-specific implementation of the class. I'd contrast these with the default values of the parameters, which I did put in the base class because we'd want to use them across different implementations of the class (such as the test fixtures, for example).